src/main/kotlin/uk/bfi/uvaudit/security/AuditUserService.kt

```diff
@@ -33,36 +35,52 @@ class AuditUserService(
         )
 
         val dbUserIds = jdbcTemplate.queryForList(sql, params, Long::class.java)
+        val userId = DataAccessUtils.uniqueResult(dbUserIds)
 
-        return when {
-            dbUserIds.isEmpty() -> {
-                null
-            }
-            dbUserIds.size == 1 -> {
-                AuditUser(dbUserIds[0], oidcUser)
-            }
-            else -> {
-                throw IncorrectResultSizeDataAccessException(0, dbUserIds.size)
-            }
+        return if (userId != null) {
+            AuditUser(userId, oidcUser)
+        } else {
+            null
         }
     }
 
     private fun createUser(oidcUser: OidcUser): AuditUser {
-        val sql = "INSERT INTO user (sub) VALUES (:sub)"
+        val sql = "INSERT INTO user (sub, email, department) VALUES (:sub, :email, :department)"
         val params = MapSqlParameterSource(
-            mapOf("sub" to oidcUser.subject)
+            mapOf(
+                "sub" to oidcUser.subject,
+                "email" to oidcUser.email,
+                "department" to extractDepartmentClaim(oidcUser)
+            )
         )
         val keyHolder = GeneratedKeyHolder()
 
         jdbcTemplate.update(sql, params, keyHolder)
 
-        return when (keyHolder.key) {
-            null -> {
-                throw EmptyResultDataAccessException(1)
-            }
-            else -> {
-                AuditUser(keyHolder.key!!.toLong(), oidcUser)
-            }
+        return AuditUser(keyHolder.key!!.toLong(), oidcUser)
+    }
+
+    private fun updateUser(oidcUser: OidcUser, auditUser: AuditUser): AuditUser {
+        val sql = "UPDATE user SET email = :email, department = :department WHERE id = :id"
+        val params = MapSqlParameterSource(
+            mapOf(
+                "email" to oidcUser.email,
+                "department" to extractDepartmentClaim(oidcUser),
+                "id" to auditUser.id
+            )
+        )
+
+        jdbcTemplate.update(sql, params)
+
+        return AuditUser(auditUser.id, oidcUser)
+    }
+
+    private fun extractDepartmentClaim(oidcUser: OidcUser): String? {
+        val customClaims = oidcUser.getClaimAsMap("https://bfi.org.uk/")
+        return if (customClaims == null) {
```

Comment:
Similar to above, you can avoid doing the null check yourself with `return customClaims?.get("department") as String?`